### PR TITLE
Project settings in json file, and some bugfixes and one hotfix

### LINKED ISSUE #363: Add dialogInitialPath to the ini file
when you open WL4Editor, then open ROM, it should go to the last folder where things were saved

### LINKED ISSUE #394: the new graphic manager will cause problem in the "Tileset import from another ROM" feature
the old code will load bg tile data and length anyway, then use the bg tile data ptr from the other ROM. but bg ptr will change after the implementation of graphic manager

### LINKED ISSUE #395: the graphic manager will crash both the WL4Editor and the game rom on the second save
the current save logic in Graphic Manager is: we remove all the bg graphic things from the ROM, then save everything back to the ROM
if you have already applied some bg pointer to some Rooms in a Level, then you use the Graphic Manager to process sacing everything again. the pointer of all the graphic things will be modified
the Rooms using the old bg data will refer to some invalidated space
if the space is not rewrite by some other data, then the Room can still be loaded correctly
but if the space is overwritten by some other data, then both the game and the WL4Editor crash, since the layer data does not exist at the old place any more.

### PR DESCRIPTION
**Pull request by shinespeciall**
Resolve #363 
Fix #394 
Fix #395

### CODE REVIEW

File: AssortedGraphicUtils.cpp [hunk 1]
```diff
@@ -582,3 +598,122 @@ QVector<ROMUtils::SaveData> AssortedGraphicUtils::CreateSaveData(AssortedGraphic
...
+/// <summary>
+/// Cehck if the current input entry can be changed or deleted.
+/// </summary>
+/// <param name="entry">
+/// The struct data of the graphic entry.
+/// </param>
+/// <returns>
+/// return true if the entry is not used in any Tilesets or Rooms.
+/// </returns>
+bool AssortedGraphicUtils::CheckEditability(AssortedGraphicEntryItem &entry, unsigned int &find_l, unsigned int &find_r, unsigned int &find_t)
```
Comment (Goldensunboy) [inline AssortedGraphicUtils.cpp:611]: What is the purpose of find_l, find_r, find_t?

File: SettingsUtils.cpp [hunk 2]
```diff
@@ -52,4 +64,122 @@ namespace SettingsUtils
...
+                jsonObj = settingJsonDoc.object();
+            }
+        }
+
+        // Load settings one by one, and send them into the runtime global variables of WL4Editor
+        QStringList list = jsonObj.keys();
+
+        //--------------------------------------------------------------------------------------------------------------
+        //--------------------------------------------------------------------------------------------------------------
+        // super long shitty logic to maintain project settings
```
Comment (Goldensunboy) [inline SettingsUtils.cpp:120]: Remove this comment

File: AssortedGraphicUtils.cpp (hunk 1 above)
Comment (shinespeciall) [reply AssortedGraphicUtils.cpp:620]: added the comment

File: SettingsUtils.cpp (hunk 2 above)
Comment (shinespeciall) [reply SettingsUtils.cpp]: removed

**Review by Goldensunboy — APPROVED** [review]

**Pull request merged**